app/models/relationship.rb

```diff
@@ -157,7 +157,8 @@ def create_or_update_parent_id
   end
 
   def self.create_unless_exists(source_id, target_id, relationship_type, options = {})
-    r = Relationship.where(source_id: source_id, target_id: target_id).where('relationship_type = ?', relationship_type.to_yaml).last
+    r = Relationship.where(target_id: target_id).where.not(source_id: source_id).last
+    r = Relationship.where(source_id: source_id, target_id: target_id).where('relationship_type = ?', relationship_type.to_yaml).last unless r.nil?
```

Comment:
Sorry @melsawy , it's still not clear to me what's the idea here (and I think that we need a comment in this code to explain it, like we had in the other parts of this method).

Reply:
Done and added more comments